### PR TITLE
3DTextShader instancing updates

### PR DESCRIPTION
**Pull request by killerantz**
I updated the 3DTextShader and a couple vertex shaders to include instancing and now works with both multi pass and single pass stereo rendering build types.  #551

### CODE REVIEW

File: Assets/HoloToolkit/UI/Shaders/3DTextShader.shader [hunk 1]
```diff
@@ -48,29 +50,39 @@
...
+					UNITY_VERTEX_INPUT_INSTANCE_ID
+					UNITY_VERTEX_OUTPUT_STEREO
+
                 };
 
                 sampler2D _MainTex;
                 float4 _MainTex_ST;
-                fixed4 _Color;
+				UNITY_INSTANCING_CBUFFER_START(Props)
+					UNITY_DEFINE_INSTANCED_PROP(fixed4, _Color)
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/UI/Shaders/3DTextShader.shader]: What's with the strange space formatting?

File: Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTriplanar.shader [hunk 2]
```diff
@@ -1,4 +1,10 @@
-﻿// Upgrade NOTE: replaced '_Object2World' with 'unity_ObjectToWorld'
+﻿// Upgrade NOTE: replaced 'UNITY_INSTANCE_ID' with 'UNITY_VERTEX_INPUT_INSTANCE_ID'
+
+// Upgrade NOTE: replaced 'UNITY_INSTANCE_ID' with 'UNITY_VERTEX_INPUT_INSTANCE_ID'
+
+// Upgrade NOTE: replaced 'UNITY_INSTANCE_ID' with 'UNITY_VERTEX_INPUT_INSTANCE_ID'
+
+// Upgrade NOTE: replaced '_Object2World' with 'unity_ObjectToWorld'
 
 Shader "HoloToolkit/UnlitTriplanar"
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTriplanar.shader]: Could we rename to `HoloToolkit/Examples/UnlitTriplanar`?

File: Assets/HoloToolkit/UI/Shaders/3DTextShader.shader (hunk 1 above)
Comment (killerantz) [reply Assets/HoloToolkit/UI/Shaders/3DTextShader.shader]: The formatting keeps getting funky, I will fix.

File: Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTriplanar.shader (hunk 2 above)
Comment (killerantz) [reply Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTriplanar.shader]: Good point, will do.

Comment (StephenHodgson) [reply Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTriplanar.shader]: Also, I think the upgrade notes are injected by Unity, we can probably remove those as well, and add the default CLA license header.

File: Assets/HoloToolkit/UI/Shaders/3DTextShader.shader [hunk 3]
```diff
@@ -46,48 +52,59 @@
         Pass
         {
             CGPROGRAM
-                #pragma vertex vert
-                #pragma fragment frag
-                #include "UnityCG.cginc"
+            #pragma vertex vert
+            #pragma fragment frag
+			#pragma multi_compile_instancing
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/UI/Shaders/3DTextShader.shader]: One more weird tab

File: Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTransparentTriplanar.shader [hunk 4]
```diff
@@ -22,45 +26,63 @@ Shader "HoloToolkit/UnlitTransparentTriplanar"
...
-		fixed3 _LightDir;
-		fixed4 _Color;
-		float1 _LightIntensity;
-		float1 _CameraIntensity;
+			fixed _Ambient;
+			fixed3 _LightDir;
+			float1 _LightIntensity;
+			float1 _CameraIntensity;
+			UNITY_INSTANCING_CBUFFER_START(Props)
+				UNITY_DEFINE_INSTANCED_PROP(fixed4, _Color)
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/Prototyping/Shaders/UnlitTransparentTriplanar.shader]: Another tab

**Pull request merged**